cmd/api/src/database/saved_queries.go

```diff
@@ -39,3 +39,18 @@ func (s *BloodhoundDB) CreateSavedQuery(userID uuid.UUID, name string, query str
 
 	return savedQuery, CheckError(s.db.Create(&savedQuery))
 }
+
+func (s *BloodhoundDB) DeleteSavedQuery(id int) error {
+	return CheckError(s.db.Delete(&model.SavedQuery{}, id))
+}
+
+func (s *BloodhoundDB) SavedQueryBelongsToUser(userID uuid.UUID, savedQueryID int) (bool, error) {
+	var savedQuery model.SavedQuery
+	if result := s.db.First(&savedQuery, savedQueryID); result.Error != nil {
+		return false, result.Error
```

Comment:
```suggestion
		return false, CheckError(result)
```
This will ensure we properly wrap the error if it's a `gorm.ErrRecordNotFound` to prevent needing to leak `gorm` outside of the database layer.